src/components/chart/state-history-chart-line.ts

```diff
@@ -141,16 +141,10 @@ export class StateHistoryChartLine extends LitElement {
                 `${context.dataset.label}: ${formatNumber(
                   context.parsed.y,
                   this.hass.locale,
-                  this.data[context.datasetIndex]?.entity_id
-                    ? getNumberFormatOptions(
-                        this.hass.states[
-                          this.data[context.datasetIndex].entity_id
-                        ],
-                        this.hass.entities[
-                          this.data[context.datasetIndex].entity_id
-                        ]
-                      )
-                    : undefined
+                  getNumberFormatOptions(
+                    undefined,
```

Comment:
Why don't we pass the state?

Reply:
I thought about this, but looking at what getNumberFormatOptions does with the state, I'm not sure it made sense to use it. 

The only thing it seems to check is if the current state is an integer. Should the current state of an entity affect how all its historical states are rendered?

Reply:
Probably not, unless we assume the state was always an integer, which I dont think we can do.
More the step that might be useful.

Reply:
I guess you could combine the current step attribute, and the historical state value, merge them into a HassEntity, and pass that to the function.